crates/bevy_reflect/src/serde/de/deserializer.rs

```diff
@@ -14,6 +14,193 @@ use crate::{
+/// #         unimplemented!()
+/// #     }
+/// # }
+/// # impl ReflectHandle {
+/// #     fn asset_type_id(&self) {
+/// #         unimplemented!()
+/// #     }
+/// # }
+/// #
+/// # struct AssetPathVisitor;
+/// # impl<'de> Visitor<'de> for AssetPathVisitor {
+/// #     type Value = ();
+/// #     fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result { unimplemented!() }
+/// # }
+/// # type AssetError = Box<dyn core::error::Error>;
+/// #[derive(Debug, Clone, Reflect)]
+/// struct MyAsset {
+///     name: String,
+///     mesh: Handle<Mesh>,
+/// }
+///
+/// fn load(
+///     asset_bytes: &[u8],
+///     type_registry: &TypeRegistry,
+///     load_context: &mut LoadContext,
+/// ) -> Result<MyAsset, AssetError> {
+///     struct HandleProcessor<'a> {
+///         load_context: &'a mut LoadContext,
+///     }
+///
+///     impl ReflectDeserializerProcessor for HandleProcessor<'_> {
+///         fn try_deserialize<'de, D>(
+///             &mut self,
+///             registration: &TypeRegistration,
+///             _registry: &TypeRegistry,
+///             deserializer: D,
+///         ) -> Result<Result<Box<dyn PartialReflect>, D>, D::Error>
+///         where
+///             D: Deserializer<'de>,
+///         {
+///             let Some(reflect_handle) = registration.data::<ReflectHandle>() else {
+///                 // we don't want to deserialize this - give the deserializer back
+///                 return Ok(Err(deserializer));
+///             };
+///
+///             let asset_type_id = reflect_handle.asset_type_id();
+///             let asset_path = deserializer.deserialize_str(AssetPathVisitor)?;
+///
+///             let handle: Handle<LoadedUntypedAsset> = self.load_context
+///                 .load()
+///                 .with_asset_type_id(asset_type_id)
+///                 .untyped()
+///                 .load_asset(asset_path);
+///             # let _: Result<_, ()> = {
+///             Ok(Box::new(handle))
+///             # };
+///             # unimplemented!()
+///         }
+///     }
+///
+///     let mut ron_deserializer = ron::Deserializer::from_bytes(asset_bytes)?;
+///     let mut processor = HandleProcessor { load_context };
+///     let reflect_deserializer =
+///         ReflectDeserializer::with_processor(type_registry, &mut processor);
+///     let asset = reflect_deserializer.deserialize(&mut ron_deserializer)?;
+///     # unimplemented!()
+/// }
+/// ```
+///
+/// [`try_deserialize`]: Self::try_deserialize
+pub trait ReflectDeserializerProcessor {
+    /// Attempts to deserialize the value which a [`TypedReflectDeserializer`]
+    /// is currently looking at, and knows the type of.
+    ///
+    /// If you've read the `registration` and want to override the default
+    /// deserialization, return `Ok(Ok(value))` with the boxed reflected value
+    /// that you want to assign this value to. The type inside the box must
+    /// be the same one as the `registration` is for.
```

Comment:
What happens if it's not?

Reply:
`FromReflect` will fail when attempting to convert the `Box<dyn PartialReflect>` into a concrete typed. Added docs and a unit test for this.